### ISSUE TITLE
database_deletion_policy and user_deletion_policy not expexted here"

### ISSUE THREAD

**Issue by mifrank-cg** [terraform-google-modules/terraform-google-sql-db#411]
### TL;DR

We used Google Provider 4.49 and SQL module 13.0.1 but the variables database_deletion_policy and user_deletion_policy were not expected as part of the module during the terraform plan.
The provider set the default value to "DELETE" but we would like to use "ABANDON" for our PostgreSQL as recommended.
Maybe you can set the default value to "ABANDON" in the postgresql module instead of using "null".

### Expected behavior

Using desired value "ABANDON" or "DELETE" but it is not an input as far as we see.

### Observed behavior

The module always used "DELETE" as value for the "database_deletion_policy" and we could not set it to "ABANDON".

### Terraform Configuration

```hcl
module "postgresql" {
  source  = "GoogleCloudPlatform/sql-db/google//modules/postgresql"
  version = "~> 13.0"

  project_id        = module.project_infra.project_id
  name              = "instance-name"
  availability_type = var.database_availability_type
  backup_configuration = {
    enabled                        = true
    start_time                     = "01:00"
    location                       = local.default_region
    point_in_time_recovery_enabled = true
    transaction_log_retention_days = "7"
    retained_backups               = 7
    retention_unit                 = "COUNT"
  }
  database_version         = "POSTGRES_14"
  create_timeout           = "30m"
  update_timeout           = "30m"
  delete_timeout           = "30m"
  deletion_protection      = true
  disk_size                = 10
  disk_type                = "pd_ssd"
  disk_autoresize          = true
  disk_autoresize_limit    = 100
  database_deletion_policy = "ABANDON"
  user_deletion_policy     = "ABANDON"
  ip_configuration = {
    authorized_networks = []
    ipv4_enabled        = false # No public IP
    private_network     = module.shared_vpc.network_self_link
    require_ssl         = false
    allocated_ip_range  = "google-managed-services-${module.shared_vpc.network_name}"
  }
  region                          = local.default_region
  zone                            = "europe-west3-a"
  tier                            = "db-custom-2-3840"
  maintenance_window_day          = 7
  maintenance_window_hour         = 5
  maintenance_window_update_track = "stable"
  enable_default_db               = false # We do not want a "default database"
  enable_default_user             = false # We do not want a "default user"
  encryption_key_name             = "customer-managed-encryption-key"
  additional_databases = [
    {
      name      = "db1"
      charset   = "UTF8"
      collation = "en_US.UTF8"
    },
    {
      name      = "db2"
      charset   = "UTF8"
      collation = "en_US.UTF8"
    }
  ]
  additional_users = [
    {
      name     = "user1"
      password = "password1"
    },
    {
      name     = "user2"
      password = "password2"
    }
  ]
  user_labels = {
    client = local.client
  }
}
```

### Terraform Version

```sh
Terraform v1.3.7
```

### Additional information

+ provider registry.terraform.io/gavinbunney/kubectl v1.14.0
+ provider registry.terraform.io/hashicorp/google v4.49.0
+ provider registry.terraform.io/hashicorp/google-beta v4.49.0
+ provider registry.terraform.io/hashicorp/helm v2.8.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.16.1
+ provider registry.terraform.io/hashicorp/local v2.3.0
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.4.3
+ provider registry.terraform.io/hashicorp/time v0.9.1

**Comment by Vierian** [comment]
Same here
